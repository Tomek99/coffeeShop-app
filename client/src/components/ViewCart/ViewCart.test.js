import { getByRole, render, screen } from "@testing-library/react";
import ViewCart from "./ViewCart";
import user from "@testing-library/user-event";
import { Context } from "../../Contexts/Context";
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom";

const items1 = [];
const items2 = [
  {
    available: "true",
    brand: "lavazza",
    createdAt: "2023-02-03T11:11:55.634Z",
    imageUrl:
      "https://res.cloudinary.com/dvoduabha/image/upload/v1679902516/coffee/lavazza/ground/4_LavazzaTierraBioOrganicforAfrica_qvc2in.png",
    intensity: "7",
    name: "Lavazza Tierra Bio-Organic for Africa",
    oldPrice: "8.00",
    origin: "Ethiopia ",
    price: "7.00",
    productRatings: [5],
    quantity: "1",
    type: "ground coffee",
    updatedAt: "2024-03-13T09:17:24.519Z",
    weight: "250 g",
    __v: 0,
    _id: "63dcebfb954a9e16291723c8",
  },
];

jest.mock("./FillCart/CartSummary/CartSummary", () => {
  return () => {
    return <div>test</div>;
  };
});

const renderComponent = (cartItems) => {
  return render(
    <Context.Provider value={{ cartItems }}>
      <MemoryRouter>
        <ViewCart />
      </MemoryRouter>
    </Context.Provider>
  );
};

describe("render components", () => {
  test("should render EmptyCart", () => {
    const { container } = renderComponent(items1);

    const div = screen.getByText("Your cart is empty");
    expect(div).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("should render one item", () => {
    const { container } = renderComponent(items2);

    const div = screen.getByText("Lavazza Tierra Bio-Organic for Africa");
    expect(div).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
