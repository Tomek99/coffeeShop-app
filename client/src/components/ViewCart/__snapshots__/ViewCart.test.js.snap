// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render components should render EmptyCart 1`] = `
<div>
  <div
    class="ViewCart"
  >
    <div
      class="emptyCart"
    >
      <header
        class="cartHeader"
      >
        Your cart is empty
      </header>
      <button
        class="BtnDefault"
      >
        Return to products page
      </button>
    </div>
  </div>
</div>
`;

exports[`render components should render one item 1`] = `
<div>
  <div
    class="ViewCart"
  >
    <div
      class="fillCart"
    >
      <div
        class="leftColumn"
      >
        <div
          class="divRow"
        >
          <header
            class="cartHeader"
          >
            Cart
            <span>
               
              (undefined)
            </span>
          </header>
          <div
            class="buttons"
          >
            <button>
              <svg
                fill="currentColor"
                height="18"
                stroke="currentColor"
                stroke-width="0"
                version="1.1"
                viewBox="0 0 16 16"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2 5v10c0 0.55 0.45 1 1 1h9c0.55 0 1-0.45 1-1v-10h-11zM5 14h-1v-7h1v7zM7 14h-1v-7h1v7zM9 14h-1v-7h1v7zM11 14h-1v-7h1v7z"
                />
                <path
                  d="M13.25 2h-3.25v-1.25c0-0.412-0.338-0.75-0.75-0.75h-3.5c-0.412 0-0.75 0.338-0.75 0.75v1.25h-3.25c-0.413 0-0.75 0.337-0.75 0.75v1.25h13v-1.25c0-0.413-0.338-0.75-0.75-0.75zM9 2h-3v-0.987h3v0.987z"
                />
              </svg>
               
              <p>
                Clear the cart
              </p>
            </button>
          </div>
        </div>
        <div
          class="items"
        >
          <div
            class="Item"
          >
            <div
              class="leftSide"
            >
              <img
                alt="https://res.cloudinary.com/dvoduabha/image/upload/v1679902516/coffee/lavazza/ground/4_LavazzaTierraBioOrganicforAfrica_qvc2in.png"
                src="https://res.cloudinary.com/dvoduabha/image/upload/v1679902516/coffee/lavazza/ground/4_LavazzaTierraBioOrganicforAfrica_qvc2in.png"
              />
              <div>
                <p>
                  Lavazza Tierra Bio-Organic for Africa
                </p>
              </div>
            </div>
            <div
              class="rightSide"
            >
              <div
                class="price"
              >
                <p>
                  $8.00
                </p>
                <p>
                  $
                  7.00
                </p>
              </div>
              <div
                class="selectMate"
              >
                <select
                  class="dropdownEl"
                >
                  <option
                    value="1"
                  >
                    1
                  </option>
                  <option
                    value="2"
                  >
                    2
                  </option>
                  <option
                    value="3"
                  >
                    3
                  </option>
                  <option
                    value="4"
                  >
                    4
                  </option>
                  <option
                    value="5"
                  >
                    5
                  </option>
                  <option
                    value="6"
                  >
                    6
                  </option>
                  <option
                    value="7"
                  >
                    7
                  </option>
                  <option
                    value="8"
                  >
                    8
                  </option>
                  <option
                    value="9"
                  >
                    9+
                  </option>
                </select>
              </div>
              <button
                class="BtnDeleteProduct"
                id="imBinIdViewCart0"
              >
                <svg
                  fill="currentColor"
                  height="18"
                  stroke="currentColor"
                  stroke-width="0"
                  version="1.1"
                  viewBox="0 0 16 16"
                  width="18"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M2 5v10c0 0.55 0.45 1 1 1h9c0.55 0 1-0.45 1-1v-10h-11zM5 14h-1v-7h1v7zM7 14h-1v-7h1v7zM9 14h-1v-7h1v7zM11 14h-1v-7h1v7z"
                  />
                  <path
                    d="M13.25 2h-3.25v-1.25c0-0.412-0.338-0.75-0.75-0.75h-3.5c-0.412 0-0.75 0.338-0.75 0.75v1.25h-3.25c-0.413 0-0.75 0.337-0.75 0.75v1.25h13v-1.25c0-0.413-0.338-0.75-0.75-0.75zM9 2h-3v-0.987h3v0.987z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="divSticky"
      >
        <div>
          test
        </div>
      </div>
    </div>
  </div>
</div>
`;
